### PR TITLE
Method getProperty() of adapters should throw exception when property doesn't exist

### PR DESCRIPTION
**Pull request by kukulich**

### CODE REVIEW

File: src/Reflection/Adapter/ReflectionClass.php [hunk 1]
```diff
@@ -184,7 +184,7 @@ public function getProperty($name)
         $betterReflectionProperty = $this->betterReflectionClass->getProperty($name);
 
         if (null === $betterReflectionProperty) {
-            return null;
+            throw new CoreReflectionException(\sprintf('Property "%s" does not exist', $name));
```
Comment (Ocramius) [inline src/Reflection/Adapter/ReflectionClass.php]: Return type hint should also be enforced, I suppose

File: src/Reflection/Adapter/ReflectionObject.php [hunk 2]
```diff
@@ -175,7 +175,7 @@ public function getProperty($name)
         $property = $this->betterReflectionObject->getProperty($name);
 
         if (null === $property) {
-            return null;
+            throw new CoreReflectionException(\sprintf('Property "%s" does not exist', $name));
```
Comment (Ocramius) [inline src/Reflection/Adapter/ReflectionObject.php]: Return type hint to be adde

File: src/Reflection/Adapter/ReflectionClass.php (hunk 1 above)
Comment (kukulich) [reply src/Reflection/Adapter/ReflectionClass.php]: There are currently no return type hints in adapters.

**Comment by kukulich** [comment]
@Ocramius Done.

**Comment by Ocramius** [comment]
@kukulich thanks, waiting for CI

**Comment by Ocramius** [comment]
🚢

**Comment by Ocramius** [comment]
Handles #389

**Pull request merged**